pkg/instrumentation/flags.go

```diff
@@ -148,8 +155,18 @@ func defaultOptions() interface{} {
 
 // configNotify is our configuration udpate notification handler.
 func configNotify(event config.Event, source config.Source) error {
-	log.Info("tracing configuration is now %v", opt.Trace)
-	Setup()
+	log.Info("instrumentation configuration is now %v", opt.Trace)
+
+	if opt.Collector == cfg.Collector || opt.Agent == cfg.Agent || opt.Metrics == cfg.Metrics {
```

Comment:
Should it be `&&` instead `||`?

Reply:
Yes, it should test if any of the endpoints have changed, so it's broken now.